src/main/java/org/cloudfoundry/promregator/lite/scrape/ScrapeControllerV2.kt

```diff
@@ -0,0 +1,65 @@
+package org.cloudfoundry.promregator.lite.scrape
+
+import org.cloudfoundry.promregator.lite.auth.AuthenticatorService
+import io.netty.channel.ChannelOption
+import io.netty.handler.timeout.ReadTimeoutHandler
+import io.netty.handler.timeout.WriteTimeoutHandler
+import io.prometheus.client.exporter.common.TextFormat
+import kotlinx.coroutines.runBlocking
+import org.cloudfoundry.promregator.lite.discovery.DiscoveryService
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.http.client.reactive.ReactorClientHttpConnector
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.reactive.function.client.*
+import reactor.netty.http.client.HttpClient
+
+private val log = mu.KotlinLogging.logger { }
+
+@RestController
+class ScrapeControllerV2(
+        private val discoverer: DiscoveryService,
+        private val webClientBuilder: WebClient.Builder,
+        private val authService: AuthenticatorService,
+) {
+    private val httpClient = HttpClient.create()
+            .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, 3000) // Connection Timeout
+            .doOnConnected { connection ->
+                connection.addHandlerLast(ReadTimeoutHandler(20)) // Read Timeout
```

Comment:
This is the new scrape endpoint. It doesn't do any metric enrichment. We can make the timeouts configurable if need be.